scrapers/bank_of_kenya.py

```diff
@@ -17,6 +17,12 @@ def bank_of_kenya_scraper():
         # TODO: Option to download n number of files
         pdf_link = page.locator("td.pdf_link > a").all()[0]
         link = pdf_link.get_attribute("href")
+
+        # Check for link validity
+        if link is None or not link.endswith(".pdf"):
+            browser.close()
+            raise ValueError("Invalid link or not found")
```

Comment:
Consider the following.

```
link = None
bool(link is None) ---> # True

link = ''
bool(link is None) ---> # False
```

In this case we want to ensure that if we don't have a link we need to stop the script. Moreover, in this case specifically, an empty string should not pass the check either. So we should do:

```suggestion
        if not link or not link.endswith(".pdf"):
            browser.close()
            raise ValueError("Invalid link or not found")
```

or if we want to go a step further:

```suggestion
        if not link or (isinstance(link, str) and not link.endswith(".pdf")):
            browser.close()
            raise ValueError("Invalid link or not found")
```

They both have pros and cons tbh. I will leave you to figure out what the second one may miss that the first one won't, and vice versa.

Though I think for all intents and purposes, let's go with the first suggestion and be more restrictive if we want.

Reply:
Thank you for the insights, I went ahead and implemented the first case, and I agree that a more strict approach can be introduced later if need be.